python/mspasspy/db/database.py

```diff
@@ -249,33 +343,26 @@ def _sync_metadata_before_update(self, d):
             d.put_string('time_standard', 'relative')
         else:
             d.put_string('time_standard', 'UTC')
-        if 'dtype' not in d:
-            if isinstance(d, TimeSeries):
-                d['dtype'] = 'TimeSeries'
-            elif isinstance(d, Seismogram):
-                d['dtype'] = 'Seismogram'
-            else:
-                raise TypeError("only TimeSeries and Seismogram are supported")
 
     def _save_history(self, mspass_object, collection='history_object'):
         history_col = self[collection]
         history_binary = pickle.dumps(ProcessingHistory(mspass_object))
         # todo jobname jobid
-        if 'history_id' in mspass_object:
+        if 'history_object_id' in mspass_object:
             # overwrite history
-            filter = {'_id': mspass_object['history_id']}
+            filter = {'_id': mspass_object['history_object_id']}
             history_col.find_one_and_replace(filter, {'nodesdata': history_binary})
         else:
             id = history_col.insert_one({'nodesdata': history_binary}).inserted_id
-            mspass_object['history_id'] = id
+            mspass_object['history_object_id'] = id
 
     def _load_history(self, mspass_object, collection='history_object'):
-        if 'history_id' not in mspass_object:
-            raise KeyError("history_id not found")
-        res = self[collection].find_one({'_id': mspass_object['history_id']})
+        if 'history_object_id' not in mspass_object:
+            raise KeyError("history_object_id not found")
+        res = self[collection].find_one({'_id': mspass_object['history_object_id']})
         mspass_object.load_history(pickle.loads(res['nodesdata']))
 
-    def _save_elog(self, data, collection='error_logs'):
+    def _save_elog(self, mspass_object, collection='error_logs'):
```

Comment:
I remember we discussed about it before, but now looking at it I think it is more rational to name the collection "elog" and its id as "elog_id", so that our schema design follows the rule of using collection name plus "_id" as the foreign id field.

Reply:
done